plugins/node/opentelemetry-instrumentation-graphql/src/utils.ts

```diff
@@ -367,12 +366,17 @@ const handleResolveSpanSuccess = (
 
 export function wrapFieldResolver<TSource = any, TContext = any, TArgs = any>(
   tracer: api.Tracer,
-  getConfig: () => Required<GraphQLInstrumentationConfig>,
+  getConfig: () => GraphQLInstrumentationParsedConfig,
```

Comment:
I'm not sure I follow the implications of this change.
Is this change why the `config.depth!` change below is necessary?

Reply:
The `config.depth!` non null assertion is by mistake. fixed 👍 

The type `Required<GraphQLInstrumentationConfig>` is not accurate as the instrumentation parses some of the config options like `depth`, `mergeItems` etc, but some where still optional, like the `enabled` and `ignoreTrivialResolveSpans` which typescript didn't like.
The new implementation makes it so the type accurately reflects which fields are nullable and which are non-null after the `DEFAULT_CONFIG` merge.